code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -370,16 +372,22 @@ mob/living/simple_animal/hostile/hitby(atom/movable/AM as mob|obj,var/speed = TH
 				src.do_attack_animation(e)
 				target_mob = e
 				change_stance(HOSTILE_STANCE_ATTACKING)
+				hostile_last_attack = world.time
 				return TRUE
-			for(var/obj/structure/window/obstacle in get_step(src, dir))
-				if(obstacle.dir == reverse_dir[dir]) // So that windows get smashed in the right order
-					obstacle.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
-					return 1
-			var/obj/structure/obstacle = locate(/obj/structure, get_step(src, dir))
-			if(istype(obstacle, /obj/structure/window) || istype(obstacle, /obj/structure/closet) || istype(obstacle, /obj/structure/table) || istype(obstacle, /obj/structure/grille))
-				obstacle.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
-				return 1
-	return 0
+
+			for(var/obj/structure/S in get_step(src, dir))
```

Comment:
If we are immediately istyping for some very specific types below, we don't need to typecheck here aswell.